autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java

```diff
@@ -37,12 +37,12 @@ public MavenVersionRepository(String baseUrl) {
                             )
                     )
         ) {
-            return new MetadataXpp3Reader()
+            return (Set<Version>) new MetadataXpp3Reader()
```

Comment:
This code can be cleaned up as well after updating the Maven dependency mentioned in the other comment.

The cast is no longer necessary.

Additional small remark: When using the Collectors, it is always best to collect the results into an immutable map (unmodifiable) instead of a standard mutable map.

            return new MetadataXpp3Reader()
                    .read(in)
                    .getVersioning()
                    .getVersions()
                    .stream()
                    .map(MavenVersion::new)
                    .collect(Collectors.toUnmodifiableSet());